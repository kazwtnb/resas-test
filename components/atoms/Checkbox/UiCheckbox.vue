<template>
  <div class="ui-checkbox" @click="onClick($event)">
    <label>
      <input type="checkbox" :checked="modelValue" />
      <div
        class="ui-checkbox-box"
        :class="{ 'ui-checkbox-box--checked': modelValue }"
      >
        <i class="fa-solid fa-check" />
      </div>
      <span>{{ label }}</span>
    </label>
  </div>
</template>
<script setup lang="ts">
interface Props {
  modelValue?: boolean;
  label: string;
}

interface Emits {
  (e: 'update:modelValue', value: boolean): void;
  (e: 'click', value: MouseEvent): void;
}

const props = withDefaults(defineProps<Props>(), {
  modelValue: false
});
const emit = defineEmits<Emits>();

const onClick = (event: MouseEvent): void => {
  event.preventDefault();
  emit('update:modelValue', !props.modelValue);
  emit('click', event);
};
</script>
<style lang="scss" scoped>
.ui-checkbox {
  display: inline-block;

  label {
    display: inline-flex;
    align-items: center;
    cursor: pointer;
  }

  input {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    width: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    pointer-events: none;
  }

  .ui-checkbox-box {
    display: inline-flex;
    width: 1rem;
    height: 1rem;
    align-items: center;
    justify-content: center;
    border: 1px solid colors.$color-border;
    border-radius: 2px;
    color: colors.$color-border;
    font-size: 0.75rem;
    transition: color 200ms ease-in, background-color 200ms ease-in;

    &:not(.ui-checkbox-box--checked) {
      > i {
        opacity: 0.3;
      }
    }

    &.ui-checkbox-box--checked {
      color: colors.$color-white;
      background-color: colors.$color-primary;
      border-color: colors.$color-primary;
    }

    + * {
      margin-left: 0.5rem;
    }
  }

  &:hover .ui-checkbox-box,
  input:focus-visible + .ui-checkbox-box {
    &:not(.ui-checkbox-box--checked) {
      background-color: colors.$color-hover;
    }

    &.ui-checkbox-box--checked {
      background-color: colors.$color-primary-hover;
      border-color: colors.$color-primary-hover;
    }
  }
}
</style>
