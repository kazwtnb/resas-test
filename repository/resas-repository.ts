import { Ref } from 'vue';
import { Population, Prefecture } from '~~/types/models/resas-model';

class ResasRepository {
  prefectures(): Ref<Prefecture[]> {
    const { data } = useFetch<Prefecture[]>('/api/resas/prefectures');
    return data;
  }

  async population(pref: Prefecture): Promise<Population | null> {
    const { data } = await useFetch<number[]>('/api/resas/population', {
      params: { prefCode: pref.prefCode }
    });

    return data.value
      ? {
        ...pref,
        data: data.value
      }
      : null;
  }
}

const $resasRepository = new ResasRepository();

export default $resasRepository;
