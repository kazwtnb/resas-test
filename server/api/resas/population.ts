import { PopulationResult, ResasResponse } from '~/types/entity/resas-entity';

export default defineEventHandler(async (event): Promise<number[]> => {
  const prefCode = useQuery(event).prefCode;
  if (!prefCode) {
    return [];
  }

  const $config = useRuntimeConfig();
  const res: ResasResponse<PopulationResult> = await $fetch(
    'api/v1/population/composition/perYear',
    {
      params: {
        prefCode: useQuery(event).prefCode,
        cityCode: '-'
      },
      baseURL: $config.public.RESAS_ENDPOINT_BASE_URL,
      headers: {
        'X-API-KEY': $config.RESAS_API_KEY
      }
    }
  );

  return (
    res?.result?.data
      ?.find((d) => d.label === '総人口')
      ?.data?.map((i) => i.value) || []
  );
});
