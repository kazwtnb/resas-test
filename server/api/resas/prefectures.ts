import { ResasResponse } from '~/types/entity/resas-entity';
import { Prefecture } from '~/types/models/resas-model';

export default defineEventHandler(async (): Promise<Prefecture[]> => {
  const $config = useRuntimeConfig();
  const res: ResasResponse<Prefecture[]> = await $fetch('api/v1/prefectures', {
    baseURL: $config.public.RESAS_ENDPOINT_BASE_URL,
    headers: {
      'X-API-KEY': $config.RESAS_API_KEY
    }
  });
  
  return res?.result;
});
