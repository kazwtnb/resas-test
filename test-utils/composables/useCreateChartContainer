export const useCreateChartContainer = () => {
  const container = document.createElement('div');
  const chartTarget = document.createElement('div');
  chartTarget.id = 'test-chart-container';
  container.appendChild(chartTarget);

  return { container, chartTarget };
};
