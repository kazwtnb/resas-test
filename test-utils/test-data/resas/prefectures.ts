import { ResasResponse } from '~~/types/entity/resas-entity';
import { Prefecture } from '~~/types/models/resas-model';

export const testPrefectures = [
  { prefCode: 1, prefName: '北海道' },
  { prefCode: 2, prefName: '青森県' },
  { prefCode: 3, prefName: '岩手県' },
  { prefCode: 4, prefName: '宮城県' },
  { prefCode: 5, prefName: '秋田県' },
  { prefCode: 6, prefName: '山形県' }
];

export const testPrefecturesResponse: ResasResponse<Prefecture[]> = {
  message: null,
  result: testPrefectures
};
