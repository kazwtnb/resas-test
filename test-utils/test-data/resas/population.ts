import { testPrefectures } from '~~/test-utils/test-data/resas/prefectures';
import {
  PopulationDataSet,
  PopulationResult,
  ResasResponse,
  YearlyPopulationItem
} from '~~/types/entity/resas-entity';
import { Population } from '~~/types/models/resas-model';

const testYearlyPopulationItems: YearlyPopulationItem[] = [
  { year: 1960, value: 4206313 },
  { year: 1965, value: 4798653 },
  { year: 1970, value: 5386163 },
  { year: 1975, value: 5923569 },
  { year: 1980, value: 6221638 },
  { year: 1985, value: 6455172 },
  { year: 1990, value: 6690603 }
];

export const testPopulationDataSet: PopulationDataSet[] = [
  { label: '総人口', data: testYearlyPopulationItems },
  { label: '年少人口', data: testYearlyPopulationItems }
];

export const testPopulationResponse: ResasResponse<PopulationResult> = {
  message: null,
  result: {
    boundaryYear: 2015,
    data: testPopulationDataSet
  }
};

export const testPopulation: Population[] = testPrefectures.map((p, index) => ({
  ...p,
  data: [...Array(10)].map((_, i) => i + index)
}));

export const createPopulation = (customize?: Partial<Population>) => {
  return {
    prefCode: 1,
    prefName: '北海道',
    data: [1, 2, 3],
    ...customize
  };
};
