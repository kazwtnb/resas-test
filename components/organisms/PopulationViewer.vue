<template>
  <div class="population-viewer">
    <PrefecturesSelect @add="add($event)" @remove="remove($event)" />
    <UiSpacer height="2rem" />
    <PopulationChart :chart-render-to="chartRenderTo" :population="population" />
  </div>
</template>
<script setup lang="ts">
import UiSpacer from '~/components/atoms/UiSpacer.vue';
import PrefecturesSelect from '~/components/molecules/PrefecturesSelect.vue';
import PopulationChart from '~/components/molecules/PopulationChart.vue';
import { usePopulation } from '~/composables/usePopulation';

interface Props {
  chartRenderTo?: HTMLDivElement;
}

const props = withDefaults(defineProps<Props>(), { chartRenderTo: undefined });

const { population, add, remove } = usePopulation();
</script>
<style lang="scss" scoped></style>
