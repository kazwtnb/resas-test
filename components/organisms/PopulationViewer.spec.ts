import { ref, nextTick } from 'vue';
import { mount } from '@vue/test-utils';
import { describe, expect, test, beforeEach, vi, beforeAll } from 'vitest';
import UiCheckbox from '../atoms/Checkbox/UiCheckbox.vue';
import PopulationViewer from './PopulationViewer.vue';
import { useCreateChartContainer } from '~~/test-utils/composables/useCreateChartContainer';
import { testPrefectures } from '~~/test-utils/test-data/resas/prefectures';
import $resasRepository from '~~/repository/resas-repository';
import { createPopulation } from '~~/test-utils/test-data/resas/population';
import { useTestConfig } from '~~/utils/chart-utils';

describe('PopulationViewer', () => {
  const testPrefecture = testPrefectures[0];
  const testPopulation = createPopulation(testPrefecture);

  beforeAll(() => {
    useTestConfig();
  });

  beforeEach(() => {
    // /serverディレクトリ配下のBFFがバグでテストで動作しないため、Repository層でモック
    // https://github.com/nuxt/framework/issues/3252
    vi.spyOn($resasRepository, 'prefectures').mockReturnValue(
      ref(testPrefectures)
    );

    vi.spyOn($resasRepository, 'population').mockReturnValue(
      new Promise((resolve) => {
        resolve(testPopulation);
      })
    );
  });

  test('チェックのつけはずしに応じてグラフが描画される', async () => {
    const { container, chartTarget } = useCreateChartContainer();

    const wrapper = mount(PopulationViewer, {
      attachTo: container,
      props: {
        chartRenderTo: chartTarget
      }
    });
    // チェックを入れる
    wrapper.findAllComponents(UiCheckbox)[0].trigger('click');
    await nextTick();

    // グラフが出る
    expect(container).toMatchSnapshot();

    // チェックを外す
    wrapper.findAllComponents(UiCheckbox)[0].trigger('click');
    await nextTick();

    // グラフが消える
    expect(container).toMatchSnapshot();
  });
});
