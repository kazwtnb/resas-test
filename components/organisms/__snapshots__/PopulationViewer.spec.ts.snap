// Vitest Snapshot v1

exports[`PopulationViewer > チェックのつけはずしに応じてグラフが描画される 1`] = `
<div>
  <div
    data-highcharts-chart="0"
    id="test-chart-container"
    style="overflow: hidden;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-0"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none;"
    >
      <svg
        class="highcharts-root"
        height="400"
        style="font-family: \\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif; font-size: 12px;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 10.1.0
        </desc>
        <defs>
          <clippath
            id="highcharts-1-"
          >
            <rect
              fill="none"
              height="345"
              width="570"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#ffffff"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          height="345"
          width="570"
          x="10"
          y="40"
        />
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="345"
          width="570"
          x="10"
          y="40"
        />
        <g
          class="highcharts-series-group"
          data-z-index="3"
        />
        <text
          class="highcharts-title"
          data-z-index="4"
          style="color: rgb(51, 51, 51); font-size: 18px; fill: #333333;"
          text-anchor="middle"
          x="300"
          y="24"
        />
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color: rgb(102, 102, 102); fill: #666666;"
          text-anchor="middle"
          x="300"
          y="10"
        />
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color: rgb(102, 102, 102); fill: #666666;"
          text-anchor="start"
          x="10"
          y="397"
        />
        <g
          class="highcharts-legend highcharts-no-tooltip"
          data-z-index="7"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            height="8"
            rx="0"
            ry="0"
            visibility="hidden"
            width="8"
            x="0"
            y="0"
          />
          <g
            data-z-index="1"
          >
            <g />
          </g>
        </g>
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor: pointer; color: rgb(153, 153, 153); font-size: 9px; fill: #999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
  <div
    data-v-app=""
  >
    <div
      class="population-viewer"
    >
      <ul
        class="ui-checkboxes"
        data-v-4ad4a939=""
      >
        
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box ui-checkbox-box--checked"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                北海道
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                青森県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                岩手県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                宮城県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                秋田県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                山形県
              </span>
            </label>
          </div>
        </li>
        
      </ul>
      <div
        class="ui-spacer"
        data-v-317ab9a8=""
      />
      <div
        class="ui-chart"
        data-v-4ef31541=""
      >
        <div
          data-v-4ef31541=""
          id="ui-chart-container"
        />
        <span
          class="ui-chart-empty-message"
          data-v-4ef31541=""
        >
          都道府県を選択するとグラフで結果が表示されます
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`PopulationViewer > チェックのつけはずしに応じてグラフが描画される 2`] = `
<div>
  <div
    data-highcharts-chart="1"
    id="test-chart-container"
    style="overflow: hidden;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-3"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none;"
    >
      <svg
        class="highcharts-root"
        height="400"
        style="font-family: \\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif; font-size: 12px;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 10.1.0
        </desc>
        <defs>
          <clippath
            id="highcharts-4-"
          >
            <rect
              fill="none"
              height="345"
              width="570"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#ffffff"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          height="345"
          width="570"
          x="10"
          y="40"
        />
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="345"
          width="570"
          x="10"
          y="40"
        />
        <g
          class="highcharts-series-group"
          data-z-index="3"
        />
        <text
          class="highcharts-title"
          data-z-index="4"
          style="color: rgb(51, 51, 51); font-size: 18px; fill: #333333;"
          text-anchor="middle"
          x="300"
          y="24"
        />
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color: rgb(102, 102, 102); fill: #666666;"
          text-anchor="middle"
          x="300"
          y="10"
        />
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color: rgb(102, 102, 102); fill: #666666;"
          text-anchor="start"
          x="10"
          y="397"
        />
        <g
          class="highcharts-legend highcharts-no-tooltip"
          data-z-index="7"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            height="8"
            rx="0"
            ry="0"
            visibility="hidden"
            width="8"
            x="0"
            y="0"
          />
          <g
            data-z-index="1"
          >
            <g />
          </g>
        </g>
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor: pointer; color: rgb(153, 153, 153); font-size: 9px; fill: #999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
  <div
    data-v-app=""
  >
    <div
      class="population-viewer"
    >
      <ul
        class="ui-checkboxes"
        data-v-4ad4a939=""
      >
        
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                北海道
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                青森県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                岩手県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                宮城県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                秋田県
              </span>
            </label>
          </div>
        </li>
        <li
          data-v-4ad4a939=""
        >
          <div
            class="ui-checkbox"
            data-v-2d44444b=""
            data-v-4ad4a939=""
          >
            <label
              data-v-2d44444b=""
            >
              <input
                data-v-2d44444b=""
                type="checkbox"
              />
              <div
                class="ui-checkbox-box"
                data-v-2d44444b=""
              >
                <i
                  class="fa-solid fa-check"
                  data-v-2d44444b=""
                />
              </div>
              <span
                data-v-2d44444b=""
              >
                山形県
              </span>
            </label>
          </div>
        </li>
        
      </ul>
      <div
        class="ui-spacer"
        data-v-317ab9a8=""
      />
      <div
        class="ui-chart"
        data-v-4ef31541=""
      >
        <div
          data-v-4ef31541=""
          id="ui-chart-container"
        />
        <span
          class="ui-chart-empty-message"
          data-v-4ef31541=""
        >
          都道府県を選択するとグラフで結果が表示されます
        </span>
      </div>
    </div>
  </div>
</div>
`;
